Sources/NIO/EventLoop.swift

```diff
@@ -580,21 +588,45 @@ extension EventLoop {
         return EventLoopIterator([self])
     }
 
-    /// Checks that this call is run from the EventLoop. If this is called from within the EventLoop this function will
-    /// have no effect, if called from outside the EventLoop it will crash the process with a trap if run in debug mode.
-    /// In release mode this function never has any effect.
+    /// Asserts that the current thread is the one tied to this `EventLoop`.
+    /// Otherwise, if running in debug mode, the process will be abnormally terminated as per the semantics of
+    /// `preconditionFailure(_:file:line:)`. Never has any effect in release mode.
     ///
     /// - note: This is not a customization point so calls to this function can be fully optimized out in release mode.
     @inlinable
-    public func assertInEventLoop(file: StaticString = #file, line: UInt = #line) {
+    public func assertInEventLoop(_ message: @autoclosure() -> String = "", file: StaticString = #file, line: UInt = #line) {
```

Comment:
This new parameter should be taken out too.

Reply:
_This_ one is wrong, and needs to be done the same way `preconditionInEventLoop` is done: add a new method.

Reply:
You're right. I didn't notice it's a new method but I think we should probably keep to the old customisation point (even the newer would be nicer). It's confusing enough that you're supposed to implement them